---
title: "SubramanianPastore_3500FinalProject"
output:
  pdf_document: default
  html_document:
    df_print: paged
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

*load packages*
```{r}
library(dplyr)
library(tidyverse)
library(DESeq2)
library(pheatmap)
library(ggrepel)
library(NOISeq)

list.files()
```

*read in count data*
```{r}
#create a empty dataframe called m to merge the data into
m = data.frame()
# using for loop read all the count files in the count_dir path
for (i in list.files(pattern = ".counts")) {
  print(paste0("reading file: ", i))
  #read file as a data frame
  
  f <- read.table(i, sep = "\t", header = TRUE)
  #rename the columns
  colnames(f) <- c("gene_id", substr(i, 1, nchar(i)-7))
  #copy the data to another dataframe called f1
  f1 <- subset(f, select= c("gene_id", substr(i, 1, nchar(i)-7)))
  
  #if the m is empty just copy the f to m
  if(length(m) == 0){
    m = f1
    
  } else 
  {
    #if the dataframe is not empty then merge the data
    m <- merge(m, f1, by.x = "gene_id", by.y = "gene_id")
  }
  rm(f1)
}

#grab the rows from the 1st colum and use it as the row-names in the dataframe
rownames(m) <- m[,1]

# remove the column-1 (gene_ids) from the data frame using dplyr::select function
m <- select(m, "P02_S109", "P08_S156", "N04_S123", "O01_S100", "B08_S56", "N06_S139", "N12_S184", "N10_S170", "P01_S101", "E03_S21", "K12_S181", "F03_S22", "B04_S26", "G12_S92", "M04_S122", "L03_S113", "O06_S140", "T02_S198", "O12_S185", "L12_S182", "G10_S76", "C05_S33", "I05_S126")
rm(f)
```

*metadata*
```{r}
Sample = c("P02_S109", "P01_S101", "B08_S56", "F03_S22", "N10_S170", "O01_S100", "K12_S181", "N04_S123", "N12_S184", "E03_S21", "N06_S139", "P08_S156", "B04_S26", "L03_S113", "L12_S182", "I05_S126", "M04_S122", "O12_S185", "C05_S33", "T02_S198", "G10_S76", "G12_S92", "O06_S140")
BodyRegion = c("Abd", "Abd", "Abd", "W3", "W3", "W3", "W2", "W2", "W2", "Pro", "Pro", "Pro", "Abd", "Abd", "Abd", "W3", "W3", "W3", "W2", "W2", "Pro", "Pro", "Pro")
Instar = c("L4", "L4", "L4", "L4", "L4", "L4", "L4", "L4", "L4", "L4", "L4", "L4", "L5", "L5", "L5", "L5", "L5", "L5", "L5", "L5", "L5", "L5", "L5")
myfactors <- data.frame(Sample, Instar, BodyRegion)
myfactors
```

*check and order metadata and data*
```{r}
# first check wheter all the columns in dataframe(m) and myfactor is present
all(colnames(m) %in% myfactors$Sample)

# Then check the order is correct
all(rownames(m) == myfactors$Sample[1])

# Then order them according to the Sample names
m <- m[, myfactors$Sample]

# Now check the order is correct after sorting 
all(colnames(m) == myfactors$Sample)
```

*length information into metadata*
```{r}
# import length information to a dataframe
df_length <- read.table("length.txt", sep = "\t", header = TRUE, row.names = 1)

# create a vector to hold the length information
mylength <- setNames(object = df_length$length, row.names(df_length))
head(mylength)
```

*create NOISeq object*
```{r}
mydata <- readData(data = m, factors = myfactors, length = mylength)
mydata

head(assayData(mydata)$exprs)
head(featureData(mydata)@data)
head(pData(mydata))
```

*Count Distribution Plot*
```{r}
countsbio = dat(mydata, factor = NULL, type = "countsbio")
explo.plot(countsbio, toplot = 1, samples = NULL, plottype = "boxplot")
```

*Length Bias Plot*
```{r}
mylenghtbias = dat(mydata, factor = "Instar", type = "lengthbias")
explo.plot(mylenghtbias, samples = NULL, toplot = "global")
```

*PCA Plots*
```{r}
myPCA = dat(mydata, type = "PCA")
explo.plot(myPCA, factor= NULL)

myPCA = dat(mydata, type = "PCA")
explo.plot(myPCA, factor= "BodyRegion")

myPCA = dat(mydata, type = "PCA")
explo.plot(myPCA, factor= "Instar")

help(explo.plot)
```

*Differential Expression*
```{r}
##INSTAR
# lc = 1 RPKM values with length correction
# lc = 0 no length correction applied
mynoiseq.bio <- noiseqbio(mydata,
                          factor = c("Instar", "BodyRegion"),
                          norm = "rpkm",
                          random.seed = 12345,
                          lc = 1)

head(mynoiseq.bio@results[[1]])

##differentially regulated genes
mynoiseq.bio.deg = degenes(mynoiseq.bio, q = 0.9 , M= NULL)

##upregulated genes in instar 4
mynoiseq.bio.deg_up = degenes(mynoiseq.bio, q = 0.9 , M= "up")
head(mynoiseq.bio.deg_up)

##downregulated genes in instar 4
mynoiseq.bio.deg_down = degenes(mynoiseq.bio, q = 0.9 , M= "down")
head(mynoiseq.bio.deg_down)

```

*Try and do reference genome pipeline on this count data*
```{r}
# Load the libraries we'll need in the following code:
library("DESeq2")
library("apeglm")
library("pheatmap")
library("tidyverse")

# create an object with the directory containing your counts:
	# !!edit this to point to your own count file directory!!
directory <- "/Users/swapnasubramanian/Documents/Hell Part 3/Spring 2021/Practical Genomics/Final Project/SubramanianPastore_3500FinalProject"

# ensure the count files are where you think they are
list.files(directory)

# look at the data frame to ensure it is what you expect:
myfactors

##converting all kallisto counts to integers so that DESeq can read it
class(m$P02_S109)

m$P02_S109 <- as.integer(m$P02_S109)
m$P01_S101 <- as.integer(m$P01_S101)
m$B08_S56 <-as.integer(m$B08_S56)
m$F03_S22 <- as.integer(m$F03_S22)
m$N10_S170 <- as.integer(m$N10_S170)
m$O01_S100 <- as.integer(m$O01_S100)
m$K12_S181 <- as.integer(m$K12_S181)
m$N04_S123 <- as.integer(m$N04_S123)
m$N12_S184 <- as.integer(m$N12_S184)
m$E03_S21 <- as.integer(m$E03_S21)
m$N06_S139 <- as.integer(m$N06_S139)
m$P08_S156 <- as.integer(m$P08_S156)
m$B04_S26 <- as.integer(m$B04_S26)
m$L03_S113 <- as.integer(m$L03_S113)
m$L12_S182 <- as.integer(m$L12_S182)
m$I05_S126 <- as.integer(m$I05_S126)
m$M04_S122 <- as.integer(m$M04_S122)
m$O12_S185 <- as.integer(m$O12_S185)
m$C05_S33 <- as.integer(m$C05_S33)
m$T02_S198 <- as.integer(m$T02_S198)
m$G10_S76 <- as.integer(m$G10_S76)
m$G12_S92 <- as.integer(m$G12_S92)
m$O06_S140 <- as.integer(m$O06_S140)


# create the DESeq data object from Matrix (I kept the name from the example to make it easier to go through the pipeline even though it isn't HTSeq data :)
ddsHTSeq <- DESeqDataSetFromMatrix(
  countData = m,
  colData = myfactors,
		design = ~ Instar + BodyRegion + Instar:BodyRegion)

# what does expression look like across genes?

# sum counts for each gene across samples
sumcounts <- rowSums(counts(ddsHTSeq))
# take the log
logsumcounts <- log(sumcounts,base=10)
# plot a histogram of the log scaled counts
hist(logsumcounts,breaks=100)

# you can see the typically high dynamic range of RNA-Seq, with a mode in the distribution around 1000 fragments per gene, but some genes up over 1 million fragments. 

# get genes with summed counts greater than 20
keep <- sumcounts > 20

# keep only the genes for which the vector "keep" is TRUE
ddsHTSeq <- ddsHTSeq[keep,]

dds <- DESeq(ddsHTSeq)

# get results table
res <- results(dds)

# get a quick summary of the table
summary(res)

# check out the first few lines
head(res)

# get shrunken log fold changes
#get the options for coef 
resultsNames(dds)

res_shrink <- lfcShrink(dds,coef="Instar_L5_vs_L4")
help("lfcShrink")


# plot the shrunken log2 fold changes against the raw changes:
plot(
	x=res$log2FoldChange,
	y=res_shrink$log2FoldChange,pch=20,
	cex=.2,
	col=1+(res$padj < 0.05),
	xlab="raw log2 fold change",
	ylab="shrunken log2 fold change"
	)
abline(0,1)

# get the top 20 genes by shrunken log2 fold change
top20 <- order(-abs(res_shrink$log2FoldChange))[1:20]
res_shrink[top20,]


plotMA(res_shrink, ylim=c(-4,4))

# negative log-scaled adjusted p-values
log_padj <- -log(res_shrink$padj,10)
log_padj[log_padj > 100] <- 100

# plot
plot(x=res_shrink$log2FoldChange,
     y=log_padj,
     pch=20,
     cex=.2,
     col=(log_padj > 10)+1, # color padj < 0.1 red
     ylab="negative log-scaled adjusted p-value",
     xlab="shrunken log2 fold changes")

# normalized, variance-stabilized transformed counts for visualization
vsd <- vst(dds, blind=FALSE)

plotPCA(vsd, intgroup="BodyRegion")
plotPCA(vsd, intgroup="Instar")

#colored by bodyregion and instar
##Abdomen clearly grouped together, L4 seems slightly grouped together but wing genes are grouped with pronotum, and L5 seems to have one individual that is an outlier
plotPCA(vsd, intgroup=c("Instar", "BodyRegion"))

# regularized log transformation of counts
rld <- rlog(dds, blind=FALSE)

# get top 50 log fold change genes of instar vs body region
top50 <- order(-abs(res_shrink$log2FoldChange))[1:50]
df <- data.frame(colData(dds)[,c("Instar", "BodyRegion")])
	rownames(df) <- colnames(dds)
	colnames(df) <- c("Instar", "BodyRegion")
pheatmap(
	assay(rld)[top50,], 
	cluster_rows=TRUE, 
	show_rownames=TRUE,
	cluster_cols=FALSE,
	annotation_col=df
	)

l2fc_ord <- order(-abs(res_shrink$log2FoldChange))
plotCounts(dds, gene=l2fc_ord[1], intgroup= c("Instar", "BodyRegion"))


```

